src/Microsoft.Framework.PackageManager/Utils/LockFileUtils.cs

```diff
@@ -151,6 +154,105 @@ private static List<string> GetPackageAssemblies(IPackage package, FrameworkName
             return results;
         }
 
+        public static LockFileLibrary CreateLockFileLibrary(
+            LocalPackageInfo packageInfo,
+            SHA512 sha512,
+            IEnumerable<NuGetFramework> frameworks,
+            string correctedPackageName = null)
+        {
+            var lockFileLib = new LockFileLibrary();
+
+            // package.Id is read from nuspec and it might be in wrong casing.
+            // correctedPackageName should be the package name used by dependency graph and
+            // it has the correct casing that runtime needs during dependency resolution.
+            lockFileLib.Name = correctedPackageName ?? packageInfo.Id;
+            lockFileLib.Version = packageInfo.Version;
+
+            using (var nupkgStream = File.OpenRead(packageInfo.ZipPath))
+            {
+                lockFileLib.Sha = Convert.ToBase64String(sha512.ComputeHash(nupkgStream));
+                nupkgStream.Seek(0, SeekOrigin.Begin);
+
+                var packageReader = new PackageReader(nupkgStream);
+                lockFileLib.Files = packageReader.GetFiles().ToList();
+
+                foreach (var framework in frameworks)
+                {
+                    var dependencies = NuGetFrameworkUtility.GetNearest(packageReader.GetPackageDependencies(),
+                        framework, item => item.TargetFramework)?.Packages.ToList();
+                    var runtimeAssemblies = NuGetFrameworkUtility.GetNearest(packageReader.GetReferenceItems(),
+                        framework, item => item.TargetFramework)?.Items.ToList();
+
+                    // The compatibility mapping "dnxcore50 -> portable-net45+win8" is missing in NuGet core libs
+                    // We need to do that check here
+                    if (runtimeAssemblies == null &&
+                        framework == FrameworkConstants.CommonFrameworks.DnxCore50)
+                    {
+                        runtimeAssemblies = packageReader.GetReferenceItems()
+                            .FirstOrDefault(x => string.Equals(x.TargetFramework.Profile, "Profile7"))?.Items.ToList();
+                    }
+
```

Comment:
@emgarten @davidfowl , I added the missing compatibility check above, please help to review. Thanks.